core/vulkan/perfetto_producer/perfetto_threadlocal_emitter.inc

```diff
@@ -0,0 +1,235 @@
+        typename PerfettoProducer<T>::TraceContext::TracePacketHandle& packet,
+        perfetto::protos::pbzero::InternedData** interned_data) {
+    uint64_t val = interned_names_.findOrZero(name);
+    if (!val) {
+        val =  interned_names_.count() + 1;
+        interned_names_[name] = val;
+        if (!*interned_data) {
+            *interned_data = packet->set_interned_data();
+        }
+        perfetto::protos::pbzero::LegacyEventName* event_name =
+        (*interned_data)->add_legacy_event_names();
+        event_name->set_iid(val);
+        event_name->set_name(name);
+        return interned_names_.count();
+    }
+    return val;
+}
+
+template<typename T>
+uint64_t ThreadlocalEmitter<T>::InternAnnotationName(
+        const char* name,
+        typename PerfettoProducer<T>::TraceContext::TracePacketHandle& packet,
+        perfetto::protos::pbzero::InternedData** interned_data) {
+    uint64_t val = interned_annotation_names_.findOrZero(name);
+    if (!val) {
+        val =  interned_annotation_names_.count() + 1;
+        interned_annotation_names_[name] = val;
+        if (!*interned_data) {
+            *interned_data = packet->set_interned_data();
+        }
+        auto* event_name = (*interned_data)->add_debug_annotation_names();
+        event_name->set_iid(val);
+        event_name->set_name(name);
+        return interned_annotation_names_.count();
+    }
+    return val;
+}
+
+
+template<typename T>
+uint64_t ThreadlocalEmitter<T>::InternCategory(
+        const char* name,
+        typename PerfettoProducer<T>::TraceContext::TracePacketHandle& packet,
+        perfetto::protos::pbzero::InternedData** interned_data) {
+        uint64_t val = interned_categories_.findOrZero(name);
+    if (!val) {
+        val =  interned_categories_.count() + 1;
+        interned_categories_[name] = val;
+        if (!*interned_data) {
+            *interned_data = packet->set_interned_data();
+        }
+        auto* event_name = (*interned_data)->add_event_categories();
+        event_name->set_iid(val);
+        event_name->set_name(name);
+        return interned_categories_.count();
+    }
+    return val;
+}
+
+template<typename T>
+void ThreadlocalEmitter<T>::ResetIfNecessary() {
+    if (reset_) {
+        interned_annotation_names_.clear();
+        interned_names_.clear();
+        emitted_thread_data_ = false;
+        emitted_process_data_ = false;
+        reset_ = false;
+    }
+}
+
+template<typename T>
+void ThreadlocalEmitter<T>::EmitThreadData() {
+    if (emitted_thread_data_) {
+        return;
+    }
+    PerfettoProducer<T>::Trace(
+        [this](typename PerfettoProducer<T>::TraceContext ctx) {
+            uint64_t time = core::GetNanoseconds();
+            auto packet = ctx.NewTracePacket();
+            packet->set_timestamp(time);
+            //perfetto::protos::pbzero::InternedData* interned_data = nullptr;
+            //uint64_t nameidd = InternName(packet, *interned_data);
```

Comment:
remove?

Reply:
Thanks, it went away, then came back. I have been using too many machines